### DISCUSSION TITLE
Secondary DNS Server option inside AGH DHCP function?

### DISCUSSION THREAD

**Discussion by rhizoma** [AdguardTeam/AdGuardHome#3698] [Q&A]
Does AGH for linux have a secondary DNS Server option inside the AGH DHCP function (as a backup should my Rpi fail or shut down)?

**Comment by timkgh** [comment]
I think you need to edit the config file and use option 6, e.g. `'6 ips 1.2.3.4,5.6.7.8'`
https://github.com/AdguardTeam/AdGuardHome/wiki/DHCP

**Comment by rhizoma** [comment]
Thanks - I'm new to this - I edited the AdGuardHome.yaml file as shown below. The intent for the secondary dns is so that if my Rpi with adguardhome fails, I still have internet access. However, when I shut down the Rpi, I still lose access. Perhaps my syntax for option 6 is wrong?

dhcpv4:
gateway_ip: 192.168.1.1
subnet_mask: 255.255.255.0
range_start: 192.168.1.110
range_end: 192.168.1.254
lease_duration: 86400
icmp_timeout_msec: 1000
options:
- 6 ips 9.9.9.9
- 149.112.112.112

**Comment by timkgh** [comment]
The syntax should be something like this `'6 ips 192.168.1.1,9.9.9.9'` (but I have not tried it). And your DHCP clients need to refresh their leases to get the updated settings.

However, what you are trying to do won't work as you expect it to. The DHCP server will return 2 DNS servers to the clients and the clients are free to use whichever one they want and switch between them. Even though they are called primary and secondary, there's no order implied, they are considered equivalent. So some DNS requests will go through your AGH, others will go directly to Quad9 and won't be filtered.

If you want redundancy, you need 2 installations of AGH, on 2 devices, configured the same way. For now you'd have to maintain the configs in sync manually, though see this issue https://github.com/AdguardTeam/AdGuardHome/issues/573

**Reply by rhizoma** [reply]
Ok, that's what I was wondering since there was no access with the Rpi off, despite tinkering with the option 6 syntax. 
Thanks very much for clarifying. I can live without redundancy for now ;-)
